WorkspaceServer.Tests/Kernel/LanguageKernelTests.cs

```diff
@@ -943,28 +944,33 @@ public async Task When_SubmitCode_command_adds_packages_to_csharp_kernel_then_th
                 .NotContain(e => e.Code.Contains("#r"));
         }
 
-        [Fact]
-        public async Task When_SubmitCode_command_adds_packages_to_csharp_kernel_then_PackageAdded_event_is_raised()
+        [Theory]
+        [InlineData(Language.CSharp, "Microsoft.Extensions.Logging.ILogger logger = null;")]
+        [InlineData(Language.FSharp, "let logger: Microsoft.Extensions.Logging.ILogger = null")]
+        public async Task When_SubmitCode_command_adds_packages_to_kernel_then_PackageAdded_event_is_raised(Language language, string expression)
         {
-            var kernel = new CompositeKernel
+            IKernel kernel = language switch
             {
-                new CSharpKernel().UseNugetDirective()
+                Language.CSharp => new CompositeKernel { new CSharpKernel().UseNugetDirective() },
+                Language.FSharp => new FSharpKernel(),
             };
 
-            var command = new SubmitCode("#r \"nuget:Microsoft.Extensions.Logging, 2.2.0\" \nMicrosoft.Extensions.Logging.ILogger logger = null;");
+            var code = $@"
+#r ""nuget:Microsoft.Extensions.Logging, 2.2.0""
+{expression}".Trim();
+            var command = new SubmitCode(code);
 
             var result = await kernel.SendAsync(command);
 
             using var events = result.KernelEvents.ToSubscribedList();
 
             events
-                .First()
                 .Should()
-                .Match(e => e is DisplayedValueProduced && ((DisplayedValueProduced)e).Value.ToString().Contains("Installing"));
```

Comment:
Since the F# kernel handles `#r "nuget:..."` natively, `DisplayedValueProduced` won't be the first event, but it _does_ eventually appear.